### PR TITLE
Ignore callbacks while closing

### PR DESCRIPTION
**Pull request by AxelJunker**
Part of https://github.com/insurello/insurello/issues/1474 (nr. 6)

Bonus: Add fantomas formatting

### CODE REVIEW

File: src/Insurello.RabbitMqClient/MqClient.fs [hunk 1]
```diff
@@ -661,9 +735,11 @@ module MqClient =
               OnConsumerCancelled =
                   fun _ ->
                       logError (null, "Got OnConsumerCancelled event", ())
-                      exit 9
+                      Async.singleton ()
+              // exit 9
```
Comment (axelo) [inline src/Insurello.RabbitMqClient/MqClient.fs]: Probably good to revert this

File: src/Insurello.RabbitMqClient/MqClient.fs [hunk 2]
```diff
@@ -661,9 +735,11 @@ module MqClient =
...
 
               OnShutdown =
                   fun _ ->
                       logError (null, "Got OnShutdown event", ())
-                      exit 9 }
+                      Async.singleton () }
+//exit 9 }
```
Comment (axelo) [inline src/Insurello.RabbitMqClient/MqClient.fs]: Probably good to revert this

File: src/Insurello.RabbitMqClient/MqClient.fs [hunk 3]
```diff
@@ -295,55 +339,70 @@ module MqClient =
...
+            model.rpcConsumer.Model.BasicConsume(
+                queue = queueName,
+                autoAck = true,
+                consumerTag = consumerTag,
+                noLocal = false,
+                exclusive = false,
+                arguments = null,
+                consumer = model.rpcConsumer
+            )
             |> ignore // Must be true for direct-reply-to
```
Comment (axelo) [inline src/Insurello.RabbitMqClient/MqClient.fs:367]: You can move this comment to row 356 (autoAck)

File: src/Insurello.RabbitMqClient/MqClient.fs [hunk 4]
```diff
@@ -197,13 +208,26 @@ module MqClient =
...
 
     let private closeChannelConsumer: Model -> unit =
-        fun (Model model) -> if model.channelConsumer.Model.IsOpen then model.channelConsumer.Model.Close() else ()
+        fun (Model model) ->
+            if model.channelConsumer.Model.IsOpen then
+                model.channelConsumer.Model.Close()
+            else
+                ()
 
     let close: Model -> unit =
```
Comment (axelo) [inline src/Insurello.RabbitMqClient/MqClient.fs:227]: We can add a fancy `/// summary` comment, similar to the comment below but changed tense.

File: .config/dotnet-tools.json [hunk 5]
```diff
@@ -0,0 +1,10 @@
+{
+  "version": 1,
+  "isRoot": true,
+  "tools": {
+    "fantomas-tool": {
+      "version": "4.7.0",
+      "commands": ["fantomas"]
+    }
+  }
+}
```
Comment (axelo) [inline .config/dotnet-tools.json:10]: Let's run in it build_and_test.yml too

File: src/Insurello.RabbitMqClient/MqClient.fs (hunk 4 above)
Comment (AxelJunker) [reply src/Insurello.RabbitMqClient/MqClient.fs:227]: Not sure what it should say

Comment (axelo) [reply src/Insurello.RabbitMqClient/MqClient.fs:224]: > We can add a fancy `/// summary` comment, similar to the comment below but changed tense.

** similar to the comment below but changed tense.**

Comment (axelo) [reply src/Insurello.RabbitMqClient/MqClient.fs:227]: Put yourself in the user of this api and write what you would have wanted to read.

Comment (AxelJunker) [reply src/Insurello.RabbitMqClient/MqClient.fs:227]: Ah, I thought you meant `closeChannelConsumer`

**Review by axelo — APPROVED** [review]
⭐️

**Pull request merged**